From 4a565ab5dfcb81116696f49ddcdeb4bcbf548a19 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 7 Jan 2025 18:11:39 +0100
Subject: [PATCH 1428/1695] drivers: stm32_bsec: add
 stm32_bsec_self_hosted_debug_is_enabled()

Add the stm32_bsec_self_hosted_debug_is_enabled() API that is used to
detect if the self-hosted debug is enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5d0c2e8ae2d627ddc517f64303c6e193bdab713e drivers: stm32_bsec: add stm32_bsec_self_hosted_debug_is_enabled()]
Change-Id: Ie8abc0e7435bd0f2242ae0e0a3b24fac0aa29469
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/427390
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_bsec.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/drivers/stm32_bsec.c b/core/drivers/stm32_bsec.c
index 73be58839..166533ee0 100644
--- a/core/drivers/stm32_bsec.c
+++ b/core/drivers/stm32_bsec.c
@@ -95,6 +95,9 @@
 #define BSEC_MODE_PWR			BIT(5)
 #define BSEC_MODE_CLOSED		BIT(8)
 
+/* BSEC_DENR register fields */
+#define BSEC_DENR_DBGSWEN		BIT(10)
+
 /* BSEC_DEBUG bitfields */
 #ifdef CFG_STM32MP13
 #define BSEC_DEN_ALL_MSK		(GENMASK_32(11, 10) | GENMASK_32(8, 1))
@@ -508,6 +511,11 @@ uint32_t stm32_bsec_read_debug_conf(void)
 	return io_read32(bsec_base() + BSEC_DEN_OFF) & BSEC_DEN_ALL_MSK;
 }
 
+bool stm32_bsec_self_hosted_debug_is_enabled(void)
+{
+	return stm32_bsec_read_debug_conf() & BSEC_DENR_DBGSWEN;
+}
+
 static TEE_Result set_bsec_lock(uint32_t otp_id, size_t lock_offset)
 {
 	uint32_t bank = otp_bank_offset(otp_id);
-- 
2.47.2

